src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -159,4 +173,8 @@ public final class {{ target_kind_name }} {
             writer.stepOut();
         {% endif %}
     }
+
+    {% for inline_type in anonymous_types -%}
```

Comment:
(🗺️ PR tour) This is where all the anonymous types will be generated for given top level type definition.

Reply:
Will this approach also work for nested types within other nested types? E.g.:
```
type::{
  name: Foo
  fields: {
    position: {
      x: {
        fields: { 
          real_part: float, 
          imaginary_part: float 
        }
      },
      y: {
        fields: { 
          real_part: float, 
          imaginary_part: float 
        }
      }
    }
  }
}
```

Reply:
Yes it does because the template defined for anonymous types also has statement to check for its anonymous types and generate it as a child class. 
[This gist](https://gist.github.com/desaikd/2c4e15f01a3f247721ce7c7db1812ea9) has generated code for your example ISL.
Here's the statement in anonymous type template that generated nested types within other nested types: https://github.com/amazon-ion/ion-cli/pull/109/files#diff-4316040659b5b43183dc6dbee67e1f131e4942f950cd2e017b7c24b352b2e51dR197